app/collections/create-form.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import Translate from 'react-translate-component';
+import apiClient from 'panoptes-client/lib/api-client';
+
+class CollectionsCreateForm extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      collectionNameLength: 0,
+      error: null
+    };
+
+    this.onSubmit = this.onSubmit.bind(this);
+    this.handleNameInputChange = this.handleNameInputChange.bind(this);
+  }
+
+  onSubmit(event) {
+    event.preventDefault();
+
+    const displayName = this.name.value;
+    const isPrivate = this.isPrivate.checked;
+
+    const links = {};
+
+    if (this.props.subjectIDs.length > 0) {
+      links.subjects = this.props.subjectIDs;
+    }
+
+    const collection = {
+      display_name: displayName,
+      private: isPrivate,
+      links
+    };
+
+    apiClient.type('collections').create(collection).save()
+      .then((newCollection) => {
+        this.name.value = '';
+        this.isPrivate.value = true;
+        this.props.onSubmit(newCollection);
+      })
+      .catch((error) => {
+        this.setState({ error });
+      });
+  }
+
+  handleNameInputChange() {
+    this.setState({
+      collectionNameLength: this.name.value.length
+    });
+  }
+
+  renderError() {
+    if (this.state.error.status === 400) {
+      return 'You can\'t name two collections the same thing!';
+    }
+
+    return 'There was a problem creating your collection.';
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.onSubmit} className="collections-create-form">
+        <div className="form-help error">
+          {this.state.error &&
+              this.renderError()}
+        </div>
+        <label>
+          <input
+            className="collection-name-input"
+            ref={(node) => { this.name = node; }}
+            onChange={this.handleNameInputChange}
+            placeholder="Collection Name"
+          />
+        </label>
+        <div className="collection-create-form-actions">
+          <label>
+            <input
+              ref={(node) => { this.isPrivate = node; }}
+              type="checkbox"
+              defaultChecked={false}
+            />
+            <Translate content="collections.createForm.private" />
+          </label>
+          <div className="submit-button-container">
+            <button type="submit" disabled={this.state.collectionNameLength < 1}>
+              <Translate content="collections.createForm.submit" />
+            </button>
+          </div>
+        </div>
+      </form>
+    );
+  }
+}
+
+CollectionsCreateForm.defaultProps = {
+  onSubmit: () => {},
+  projectID: '',
```

Comment:
This default prop should probably be null since it's set as null in the parent if there's no project.

Reply:
React might warn about that being the wrong type, because `null` is an Object and it looks like the prop should be a string.

Reply:
Hm, that's probably why it got left out of the default prop definitions originally. Regardless, we should be consistent with types used. Either default it to null and define the allowed proptypes to allow string and null or the parent needs to be edited to set it as an empty string when there's no project.

Reply:
My preference is for this to be consistently defined as a string everywhere that it appears. Allowing object or string opens up the possibility of uncaught errors if someone accidentally passes the project object when they mean to use the project ID.